Rakefile

```diff
@@ -95,6 +95,12 @@ Rake::TestTask.new do |t|
   t.test_files = test_files
 end
 
+# We need to enable the OpenSSL 3.0 legacy providers for our test suite
```

Comment:
Why do you need legacy providir ?

Reply:
There's a whole lot of tests that use legacy algorithms, I don't recall the specifics yet but you can just comment out this line and watch the failure count blow up ;)